lib/src/command/outdated.dart

```diff
@@ -111,55 +122,83 @@ class OutdatedCommand extends PubCommand {
 
     Future<_PackageDetails> analyzeDependency(PackageRef packageRef) async {
       final name = packageRef.name;
-      final current = (entrypoint.lockFile?.packages ?? {})[name]?.version;
-      final source = packageRef.source;
-      final available = (await cache.source(source).doGetVersions(packageRef))
-          .map((id) => id.version)
-          .toList()
-            ..sort(argResults['pre-releases'] ? null : Version.prioritize);
-      final upgradable = upgradablePackages
-          .firstWhere((id) => id.name == name, orElse: () => null)
-          ?.version;
-      final resolvable = resolvablePackages
-          .firstWhere((id) => id.name == name, orElse: () => null)
-          ?.version;
-      final latest = available.last;
-      final description = packageRef.description;
+      final current = (entrypoint.lockFile?.packages ?? {})[name];
+
+      final upgradable = upgradablePackages.firstWhere((id) => id.name == name,
+          orElse: () => null);
+      final resolvable = resolvablePackages.firstWhere((id) => id.name == name,
+          orElse: () => null);
+
+      // Find the latest version, and if it's overridden
+      var latestIsOverridden = false;
+      // If present as a dependency of dev_dependency we use this ref
+      var latest = await _getLatest(rootPubspec.dependencies[name]?.toRef());
+      latest ??= await _getLatest(rootPubspec.devDependencies[name]?.toRef());
+      // If not overridden in current resolution we can use this ref
+      if (!entrypoint.root.pubspec.dependencyOverrides.containsKey(name)) {
```

Comment:
Maybe start with this, this is the simplest case...

Reply:
yes, but `rootPubspec.dependencies` felt more right..